src/boosting/gbdt.cpp

```diff
@@ -249,19 +249,18 @@ void GBDT::Train(int snapshot_freq, const std::string& model_output_path) {
   }
 }
 
-void GBDT::RefitTree(const std::vector<std::vector<int>>& tree_leaf_prediction) {
-  CHECK_GT(tree_leaf_prediction.size(), 0);
-  CHECK_EQ(static_cast<size_t>(num_data_), tree_leaf_prediction.size());
-  CHECK_EQ(static_cast<size_t>(models_.size()), tree_leaf_prediction[0].size());
+void GBDT::RefitTree(const int* tree_leaf_prediction, const int nrow, const int ncol) {
+  CHECK_EQ(num_data_, nrow);
```

Comment:
Let's add the previous check as well
```suggestion
  CHECK_GT(nrow * ncol, 0);
  CHECK_EQ(num_data_, nrow);
```